### DISCUSSION TITLE
Cannot silence - Thanos

### DISCUSSION THREAD

**Discussion by sebv004** [prymitive/karma#5578] [General]
Hi guys,
I'm doing a poc with Karma and I'm able to display alarms.
But I cannot silence alarms I got such kind of errors:
`
level=error msg="History query failed" error="failed to query Prometheus for label names: Post \"prometheus-thanos-query-frontend.prometheus.svc.cluster.local:9090\": unsupported protocol scheme \"prometheus-thanos-query-frontend.prometheus.svc.cluster.local\"" labels={} uri=prometheus-thanos-query-frontend.prometheus.svc.cluster.local:9090 worker=9
level=error msg="History query failed" error="failed to query Prometheus for label names: Post \"prometheus-thanos-query-frontend.prometheus.svc.cluster.local:9090\": unsupported protocol scheme \"prometheus-thanos-query-frontend.prometheus.svc.cluster.local\"" labels={"cluster":"devtest03"} uri=prometheus-thanos-query-frontend.prometheus.svc.cluster.local:9090 worker=11
`

Here is the conf:
`
containers:
    - name: karma
      image: ghcr.io/prymitive/karma:v0.115
      ports:
        - name: http
          containerPort: 8080
          protocol: TCP
      env:
        - name: ALERTMANAGER_URI
          value: http://alertmanager-operated.prometheus.svc.cluster.local:9093
`

Could you help me please ?

**Comment by prymitive** [comment]
`msg="History query failed"` has nothing to do with silences, you get that error because you're alerts seem to be coming from Thanos, not Prometheus, so you might need to disable history - https://github.com/prymitive/karma/blob/main/docs/CONFIGURATION.md#alert-history

**Comment by sebv004** [comment]
here is my conf: https://gist.github.com/sebv004/0f35de991a2dc38c7c420736d627317c
When I submit the silence I get "Failed to fetch".
I don't see any revelant logs in the Karma pod ...

**Comment by prymitive** [comment]
`Failed to fetch` means your browser cannot connect to alertmanager, you probably want `proxy: true`  - https://github.com/prymitive/karma/blob/main/docs/CONFIGURATION.md#alertmanagers

**Reply by IGitYou** [reply]
I have this same issue, felt this was the solution, but am in a Docker container and unable to mount the karma.yaml or specify a good directory that would be the current directory it looks in. Is there a way to pass this as an ENV variable or can you let me know what the current working directory is of the container? 

Edit: I sorted this by using the $HOME variable for both my mount and my config file parameter.  Still documenting what the default working directory on the docker image or enabling and documenting the ability to pass it in stack/compose friendly means would have helped me a lot in this quest.  The post above did confirm my suspicions, that by default the failed to fetch error was related to enabling the proxy config. Thanks.

**Reply by Cillu** [reply]
Thanks to this comment, I was able to solve my karma.yaml mounting issues, cheers!

**Comment by sebv004** [comment]
Thank you so much, it works like a charm !
